blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScmContent.java

```diff
@@ -0,0 +1,84 @@
+package io.jenkins.blueocean.blueocean_github_pipeline;
+
+import io.jenkins.blueocean.commons.ErrorMessage;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Vivek Pandey
+ */
+public class GithubScmContent {
+    private final String owner;
+    private final String repo;
+    private final String path;
+    private final String message;
+    private final String base64Data;
+    private final String branch;
+    private final String sha;
+    private final boolean autoCreateBranch;
+
+    @DataBoundConstructor
+    public GithubScmContent(String owner, String repo, String path, String message, String base64Data, String sha, String branch, Boolean autoCreateBranch) {
+        this.owner = owner;
+        this.repo = repo;
+        this.path = path;
+        this.message = message;
+        this.base64Data = base64Data;
```

Comment:
Its' echoing GitHub save request and response and did not want to get in to business of transforming messages talking to GitHub apis. Let me think over it.